scripts/validations.py

```diff
@@ -0,0 +1,128 @@
+import itertools
+import json
+import sys
+
+ACTION_CONTEXT = 'context'
+ACTION_COUNT = 'count'
+ACTIONS = (ACTION_CONTEXT, ACTION_COUNT)
+
+UNLISTED_ADDONS = set()
+
+
+def parse_validations(results):
+    return (json.loads(result) for result in results)
+
+
+def unlisted_validations(results, unlisted_addons=None):
+    if unlisted_addons is None:
+        unlisted_addons = UNLISTED_ADDONS
+    return (result
+            for result in results
+            if (not result['metadata'].get('listed', True)
+                or result['metadata'].get('id') in unlisted_addons))
+
+
+def severe_validations(results):
+    return (result
+            for result in results
+            if (result['signing_summary']['high'] > 0 or
+                result['signing_summary']['medium'] > 0 or
+                result['signing_summary']['low'] > 0))
+
+
+def error_messages(results):
+    return ({'addon': result['metadata'].get('id', 'unknown-addon'),
```

Comment:
I'd rather just filter out results without IDs before we get to this step, and assume that the property exists here. If there's no ID, something is wrong, and the results aren't meaningful.